blockly-core/core/ui/fields/field_parameter.js

```diff
@@ -74,10 +74,16 @@ Blockly.FieldParameter.dropdownChange = function(text) {
       Blockly.Variables.renameVariable(oldVar, text, this.sourceBlock_.blockSpace);
     }
   } else if (text === Blockly.Msg.DELETE_PARAMETER) {
-    var result = window.confirm(Blockly.Msg.DELETE_PARAMETER_TITLE.replace('%1', oldVar));
-    if (result) {
-      Blockly.Variables.deleteVariable(oldVar, this.sourceBlock_.blockSpace);
-    }
+    Blockly.showSimpleDialog({
+      bodyText: Blockly.Msg.DELETE_PARAMETER_TITLE.replace('%1', oldVar),
+      cancelText: Blockly.Msg.DELETE,
+      confirmText: Blockly.Msg.KEEP,
+      onConfirm: null,
+      onCancel: function() {
```

Comment:
Doesn't look like it. Clicking the close button or outside invokes neither callback.

Every use of showSimpleDialog I've seen swaps the cancel and confirm buttons. It might be a good idea to swap the property names, or just rename them `leftButton` and `rightButton` to make it less confusing.